### PR TITLE
Add sample performance project

### PR DESCRIPTION
**Pull request by rbro112**
Add a simple sample project showcasing an example of Emerge perf testing.

This sample:
- Provides a sample for how to configure the gradle plugin for perf & general perf module structure.
- Shows a basic `@EmergeStartupTest` test scenario using a simple custom app link.

### CODE REVIEW

**Comment by rbro112** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11** <a href="https://app.graphite.dev/github/pr/EmergeTools/emerge-android/11" target="_blank">Graphite</a> 
    * **PR #12** <a href="https://app.graphite.dev/github/pr/EmergeTools/emerge-android/12" target="_blank">Graphite</a> 
      * **PR #13** <a href="https://app.graphite.dev/github/pr/EmergeTools/emerge-android/13" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/EmergeTools/emerge-android/13?utm_source=stack-comment).

File: .github/workflows/performance-checks.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - performance/**
```
Comment (trevor-e) [inline .github/workflows/performance-checks.yml:11]: Just want to double-check whether this should be scoped to `performance/samples/**`.

File: .github/workflows/performance-checks.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDKs
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: 17
+      - name: Run unit tests
+        run: ./gradlew :performance:performance:test -Pandroid.useAndroidX=true
```
Comment (trevor-e) [inline .github/workflows/performance-checks.yml:39]: Oh actually is this just running unit tests for the perf project? Maybe not then.

**Review by trevor-e — APPROVED** [review]

File: .github/workflows/performance-checks.yml (hunk 2 above)
Comment (rbro112) [reply .github/workflows/performance-checks.yml:40]: Correct, runs for the perf project, I intend to add lots more tests and a sanity build check in the future.

**Pull request merged**